sql/sqlwire/sql_api.proto

```diff
@@ -56,27 +56,48 @@ message SQLResponseHeader {
 }
 
 message Datum {
-  option (gogoproto.onlyone) = true;
-  oneof value {
-    bool bval = 1;
-    int64 ival = 2;
-    double dval = 3;
-    bytes blobval = 4;
+  enum Kind {
+    NULL = 0;
+    // Fixed width, big-endian. Also used for boolean values.
+    INT64 = 1;
+    // Fixed width, big-endian.
+    FLOAT64 = 2;
+    // 8-byte big-endian seconds (since Jan 1, 1970) + 4-byte big-endian
+    // nanoseconds.
+    TIME = 3;
+    // Raw bytes.
+    BYTES = 4;
```

Comment:
This will be awkward to use for clients that use JSON instead of a real protobuf implementation. In JSON output, all byte fields are base64-encoded (incidentally, this is a reason to support both string and bytes types: they're identical in proto binary format, but where character data is intended strings have a more convenient json representation), so the client will need to base64-decode and then convert to a native data type. This may be difficult to do since one reason we expect some clients to use JSON is that some languages have limited support for handling raw binary data (does this apply to any language anyone cares about except javascript in the browser?). 

Random flashback: In VBA for Excel (circa 2000), the only way to get from bytes to a floating point value was when reading from a file, and then only little-endian. I needed to import some data from a big-endian binary file into an excel spreadsheet and the only way to parse the floats was to read the file as a byte array, swap the bytes, write them back out to a temporary file and re-read them as floats.